x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/topmetrics/TopMetricsAggregationBuilder.java

```diff
@@ -47,34 +47,35 @@
                 if (size < 1) {
                     throw new IllegalArgumentException("[size] must be more than 0 but was [" + size + "]");
                 }
-                MultiValuesSourceFieldConfig metricField = (MultiValuesSourceFieldConfig) args[2];
-                return new TopMetricsAggregationBuilder(name, sorts, size, metricField);
+                @SuppressWarnings("unchecked")
+                List<MultiValuesSourceFieldConfig> metricFields = (List<MultiValuesSourceFieldConfig>) args[2];
+                return new TopMetricsAggregationBuilder(name, sorts, size, metricFields);
             });
     static {
         PARSER.declareField(constructorArg(), (p, n) -> SortBuilder.fromXContent(p), SORT_FIELD,
                 ObjectParser.ValueType.OBJECT_ARRAY_OR_STRING);
         PARSER.declareInt(optionalConstructorArg(), SIZE_FIELD);
         ContextParser<Void, MultiValuesSourceFieldConfig.Builder> metricParser = MultiValuesSourceFieldConfig.PARSER.apply(true, false);
-        PARSER.declareObject(constructorArg(), (p, n) -> metricParser.parse(p, null).build(), METRIC_FIELD);
+        PARSER.declareObjectArray(constructorArg(), (p, n) -> metricParser.parse(p, null).build(), METRIC_FIELD);
     }
 
     private final List<SortBuilder<?>> sortBuilders;
-    // TODO MultiValuesSourceFieldConfig has more things than we support and less things than we want to support
     private final int size;
-    private final MultiValuesSourceFieldConfig metricField;
+    private final List<MultiValuesSourceFieldConfig> metricFields;
+    // TODO MultiValuesSourceFieldConfig has more things than we support and less things than we want to support
```

Comment:
Can you expand on this a bit, and/or open a ticket for it? Feel free to ignore this comment if you're planing to resolve this TODO soon, but I'm worried that if we find this again in a few months we won't remember what it is we intended to do here.
Also, it looks like later on, we just turn this into a list of `ValuesSourceConfig`.  Why not just start with `ValuesSourceConfig` at this point?

Reply:
I inherited this from @polyfractal's original implementation. I can look into changing to directly to `ValuesSourceConfig`.

Reply:
ha, that's exactly the kind of problem I want to prevent ;)

We could leave this as a TODO for now, since we're hoping to merge the new Values Source work soon-ish, and that dramatically reduces the complexity of `ValuesSourceConfig`, making it more like an actual config object and less like a magic box.

Reply:
(but if we do leave it as a todo, we should still clarify what we want to do and when)